src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -146,7 +146,9 @@ public function loadChoicesForValues(array $values, $value = null)
 
         // Optimize performance in case we have an object loader and
         // a single-field identifier
-        if (null === $value && !$this->choiceList && $this->objectLoader && $this->idReader->isSingleId()) {
+        $optimize = null === $value || $isDefaultValue = is_array($value) && isset($value[0]) && $value[0] instanceof IdReader;
```

Comment:
https://github.com/kimlai/symfony/commit/aa3425e76b6f77633fdd1ace5bd582acf415a3ac  This is perfect, thanks!

I used `$isDefaultValue` simply to name the boolean, because I thought that it was hard to understand that we were testing to see if `$value` was the default `choice_value`, but your check makes much more sense, and so the need for a name is gone.